lib/common/components/ExportPatternsModal.js

```
@@ -0,0 +1,69 @@
+// @flow
+
+import React from 'react'
+import { Modal, Button } from 'react-bootstrap'
+
+import {getComponentMessages} from '../../common/util/config'
+import * as versionActions from '../../manager/actions/versions'
+import type { Feed, Snapshot } from '../../types'
+import type { ItemProps } from '../../editor/components/EditorFeedSourcePanel'
+
+type Props = {
+  createFeedVersionFromSnapshot: typeof versionActions.createFeedVersionFromSnapshot,
+  feedSource: Feed,
+}
+
+type State = {
+  showModal: boolean,
+  snapshot: ?Snapshot
+}
+
+export default class ExportPatternsModal extends React.Component<Props, State> {
+  messages = getComponentMessages('ExportPatternsModal')
+
+  state = {
+    showModal: false,
+    snapshot: null
+  }
+
+  publish (publishProprietaryFiles: boolean) {
+    const { createFeedVersionFromSnapshot, feedSource } = this.props
+    this.state.snapshot && createFeedVersionFromSnapshot(feedSource, this.state.snapshot.id, publishProprietaryFiles)
```

Are we worried about this failing silently if snapshot is null?

---

There is another comment in this file explaining how it is triggered by a snapshot class.